node/resolvers/catalog/index.ts

```diff
@@ -67,13 +67,36 @@ export const queries = {
 
   products: async (_, data, {vtex: ioContext}: ColossusContext, info) => {
     const queryTerm = data.query
+    const isAvailable = data.available
     if (test(/[\?\&\[\]\=\,]/, queryTerm)) {
-      throw new ResolverError(`The query term: '${queryTerm}' contains invalid characters.`, 500)
+      throw new ResolverError(
+        `The query term: '${queryTerm}' contains invalid characters.`,
+        500
+      )
     }
     const url = paths.products(ioContext.account, data)
-    const {data: products} = await axios.get(url, { headers: withAuthToken()(ioContext) })
+    const { data: products } = await axios.get(url, {
+      headers: withAuthToken()(ioContext),
+    })
+    let productsFiltered = products
+
+    if (isAvailable) {
```

Comment:
Não acho uma boa fazer esse tratamento aqui, de certa forma vai retornar um resultado errado para o usuário.

Exemplo: 
- na query de products peço os 4 primeiros produtos disponívels
- o resolver faz o request para o rest pedindo os 4 primeiros elementos (nesses 4 podem vir por exemplo 2 disponíveis e os outros 2 não, sendo que podem existir outros 100 produtos disponívels mais a frente)
- o algoritmo faz o filter e retorna para o front-end somente esses 2 elementos
- o que é um resultado errado pois eu pedi 4 disponíveis e só me retornou 2, existem outros dois que poderiam estar dentro do retorno, mas não foram retornados na requisição do REST

existe um filtro de disponibilidade, dá uma olhada -> https://documenter.getpostman.com/view/845/search-103/Hs43#e0ec3923-faf7-5049-8634-f6d617ce5f67

Reply:
This filter doesn't work as you are expecting! Example, the product can be available in one salesChannel but have 0 items in stock.